src/v/raft/heartbeat_manager.cc

```diff
@@ -74,12 +74,8 @@ heartbeat_manager::heartbeat_requests heartbeat_manager::requests_for_range() {
         }
 
         for (auto& [id, follower_metadata] : r->_fstats) {
-            if (follower_metadata.has_inflight_appends()) {
-                vlog(r->_ctxlog.trace, "[{}] heartbeat suppressed", id);
-                continue;
-            }
             if (
-              follower_metadata.last_sent_append_entries_req_timestamp
+              follower_metadata.last_received_reply_timestamp
```

Comment:
it seems to me that this condition is incorrect as effectively this will lead to sending two time less heartbeats. In the case when the raft group is idle and only heartbeats are exchanged between the leader and followers the `last_received_reply_timestamp` is always greater than last heartbeat for the next round after it was send

Reply:
> last_received_reply_timestamp is always greater than last heartbeat for the next round after it was send

hmm, I didn't fully follow this, how?

say at t - `last_received_reply_timestamp` is updated from the last lw hb response [here](https://github.com/redpanda-data/redpanda/blob/dev/src/v/raft/heartbeat_manager.cc#L504)

Next hb round is [guaranteed](https://github.com/redpanda-data/redpanda/blob/dev/src/v/raft/heartbeat_manager.cc#L695) to happen after `t` right? 

t_next_hb = t + delta + 150 ms, so t < t + delta (which is the if check), no?

Reply:
t3 = t2 + 150ms I think  (which is also the gist of my comment above) because of

```
void heartbeat_manager::dispatch_heartbeats() {
    ssx::background = ssx::spawn_with_gate_then(_bghbeats, [this] {
                          return _lock.with([this] {
                              return do_dispatch_heartbeats().finally([this] {
                                  if (!_bghbeats.is_closed()) {
                                      _heartbeat_timer.arm(
                                        next_heartbeat_timeout());
                                  }
                              });
                          });
                      }).handle_exception([](const std::exception_ptr& e) {
        vlog(hbeatlog.warn, "Error dispatching heartbeats - {}", e);
    });
```

Reply:
chatted offline.

Reply:
I also verified that this logic works with logging in a steady state  for a partition.. 

```
TRACE 2024-06-27 10:22:09,383 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:118 - [{id: 2, revision: 24}] full heartbeat
TRACE 2024-06-27 10:22:09,534 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:09,686 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:09,837 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:09,988 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:10,139 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:10,291 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:10,442 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:10,596 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:10,748 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:10,899 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
TRACE 2024-06-27 10:22:11,050 [shard 0:main] raft - [group_id:1, {kafka/foo/0}] heartbeat_manager.cc:107 - [{id: 2, revision: 24}] lw heartbeat
```